pkg/kubelet/cm/pod_container_manager_linux.go

```diff
@@ -91,8 +91,10 @@ func (m *podContainerManagerImpl) EnsureExists(pod *v1.Pod) error {
 		}
 		// Dispatch events to subscribed event handlers, if any.
 		// TODO(CD): Ensure it's ok to do this before calling applyLimits
-		if reply, err = m.eventDispatcher.PreStartPod(pod, string(podContainerName)); err != nil {
-			return fmt.Errorf("failed to execute PreStartPod hook for: %v: %v", pod.Name, err)
+		if m.eventDispatcher != nil {
+			if reply, err = m.eventDispatcher.PreStartPod(pod, string(podContainerName)); err != nil {
```

Comment:
one option would be to create `eventDispatcherNoop ` ( simmilar to `podContainerManagerNoop`)
and not check `if m.eventDispatcher != nil ` at all

@ConnorDoyle what do you think ?